### DISCUSSION TITLE
OG MPV time

### DISCUSSION THREAD

**Discussion by ghost** [tomasklaen/uosc#570] [Ideas]
would it be possible to implement old clickable time to get more precision time info ? MM:SS -> becomes MM:SS.XXX
it was very useful when I wanted to sync different files and it's audio tracks :(

**Comment by tomasklaen** [comment]
That's what [`time_precision` option](https://github.com/tomasklaen/uosc/blob/442fe6b2e566dbf1e60f97e722bd3917c0c80156/script-opts/uosc.conf#L181) is for.

**Comment by ghost** [comment]
thats cool but even with chatgpt i came up with lua script that would change the value but it's useless since you need to restart the mpv to see the change:

-- Define the path to the uosc.conf file
local confPath = mp.command_native({"expand-path", "~~/script-opts/uosc.conf"})

-- Function to toggle the value
local function toggleTimePrecision()
  local file = io.open(confPath, "r") -- Open the file in read mode
  local lines = {}
  local found = false

  -- Read each line in the file
  for line in file:lines() do
    if line:find("time_precision=") then
      -- Toggle the value
      found = true
      if line:find("time_precision=0") then
        line = line:gsub("time_precision=0", "time_precision=3")
      else
        line = line:gsub("time_precision=3", "time_precision=0")
      end
    end
    table.insert(lines, line) -- Store each line in a table
  end

  file:close() -- Close the file

  if not found then
    -- If the line is not found, add it to the end of the file with the default value
    table.insert(lines, "time_precision=0")
  end

  file = io.open(confPath, "w") -- Open the file in write mode

  -- Write the modified lines back to the file
  for _, line in ipairs(lines) do
    file:write(line .. "\n")
  end

  file:close() -- Close the file
end

-- Toggle the value when the 'O' key is pressed
mp.add_key_binding("O", "toggle_time_precision", toggleTimePrecision)

**Reply by christoph-heinrich** [reply]
Don't edit config files with a script.
Use [`script-opts`](https://mpv.io/manual/master/#options-script-opts) to change the options of scripts at runtime. Scripts have to support doing that and we currently don't.
I'll make a quick PR for initial support.
